From bc0c9ac27611c2f17356548b64fe2ebf080afbda Mon Sep 17 00:00:00 2001
From: Corey Bryant <corey.bryant@canonical.com>
Date: Fri, 27 Jan 2023 17:03:17 -0500
Subject: [PATCH 13/86] Revert "Bug#34823313 memleak in mysqlrouter_passwd with
 openssl 1.0.2"

This reverts commit 56f198aea2a81cce1deb0ed5cafe7f3cd5706fba.
---
 router/src/http/src/digest.h | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/router/src/http/src/digest.h b/router/src/http/src/digest.h
index 8beddad8231..86958a3059c 100644
--- a/router/src/http/src/digest.h
+++ b/router/src/http/src/digest.h
@@ -73,15 +73,7 @@ class Digest {
    *
    * Allows reused of the Digest function without reallocating memory.
    */
-  void reinit() {
-#if defined(OPENSSL_VERSION_NUMBER) && \
-    (OPENSSL_VERSION_NUMBER >= ROUTER_OPENSSL_VERSION(1, 1, 0))
-    EVP_MD_CTX_reset(ctx_.get());
-#else
-    EVP_MD_CTX_cleanup(ctx_.get());
-#endif
-    EVP_DigestInit(ctx_.get(), Digest::get_evp_md(type_));
-  }
+  void reinit() { EVP_DigestInit(ctx_.get(), Digest::get_evp_md(type_)); }
 
   /**
    * update Digest.
-- 
2.37.2

